doc/developers/development_setup.rst

```diff
@@ -330,6 +330,20 @@ the required packages.
               pytest pytest-cov ruff==0.11.2 mypy numpydoc \
               joblib threadpoolctl
 
+
+.. _pre_commit:
+
+Set up pre-commit
+^^^^^^^^^^^^^^^^^
+
+Additionally, install `pre-commit <https://pre-commit.com/#install>`__ with `pip` in your
+environment to automatically run code style checks before each commit:
+
+.. prompt::
+
+  pip install pre-commit
```

Comment:
Maybe add `pre-commit` in the `conda` / `pip` command creating the dev environment?